File: network.py
def solution(n, computers):
    answer = 0
    visited = [0]*n
    network = 1
    for i in range(n):
        DFS(i, visited, computers, n, network)
    
    answer = network
    print(visited[n-1])
    return answer

def DFS(idx, visited, computers, n, network):
    if idx == n:
        return
    if visited[idx] != 0:
        return
    visited[idx] = network
    for i in range(n):
        if visited[i] == 0 and computers[idx][i]:
            DFS(i, visited, computers, n, network)
        

File: Network.swift
import Foundation

func solution(_ n:Int, _ computers:[[Int]]) -> Int {
    var visited = [Bool](repeating: false, count: n)
    var answer = 0
    
    for i in 0..<n {
        if visited[i] == false {
            answer += 1
            dfs(&visited, n, computers, i)
        }
    }
    return answer
}

func dfs(_ visited: inout [Bool], _ n: Int, _ computers: [[Int]], _ index: Int ) {
    visited[index] = true
    for i in 0..<n {
        if visited[i] == false && computers[index][i] == 1 {
            dfs(&visited, n, computers, i)
        }
    }
    
}